### PR TITLE
Added attachment download loading indicator

### PR DESCRIPTION
**Pull request by mananjadhav**
### Details

- Adds a loading indicator for the download activity of Attachments

### Fixed Issues

$ https://github.com/Expensify/App/issues/6253

### Tests
1. Tested downloads for all platforms
2. Tested parallel downloads for the attachments

### QA Steps
1. Go to any chat with attachments (or add a new doc/zip/xlsx, etc.)
2. Click on the download icon
3. You should see an indicator while the file is about to be downloaded
4. Once the download is about to complete indicator should be replaced back with download icon.

### Tested On

- [x] Web
- [x] Mobile Web
- [x] Desktop
- [x] iOS
- [x] Android

#### Web

web-download-indicator

#### Mobile Web

mweb-download-indicator

#### Desktop

desktop-download-indicator

#### iOS

https://user-images.githubusercontent.com/3069065/143524811-22209eb7-1610-41dd-964e-a4b578e7290a.mov

#### Android

android-download-indicator

### CODE REVIEW

**Comment by mananjadhav** [comment]
@Jag96 Any updates for me on this one?

File: src/components/AttachmentView.js [hunk 1]
```diff
@@ -63,13 +68,23 @@ const AttachmentView = (props) => {
...
-            {props.shouldShowDownloadIcon && (
+            {!props.shouldShowLoadingSpinnerIcon && props.shouldShowDownloadIcon && (
                 <View style={styles.ml2}>
                     <Tooltip text={props.translate('common.download')}>
                         <Icon src={Expensicons.Download} />
                     </Tooltip>
                 </View>
             )}
+            {props.shouldShowLoadingSpinnerIcon && (
+            <View style={styles.ml2}>
```
Comment (Jag96) [inline src/components/AttachmentView.js:79]: Add indent here

File: src/libs/fileDownload/index.js [hunk 2]
```diff
@@ -5,38 +5,43 @@ import getAttachmentName from './getAttachmentName';
...
  */
 export default function fileDownload(url, fileName) {
-    fetch(url)
-        .then(response => response.blob())
-        .then((blob) => {
+    return new Promise((resolve) => {
+        fetch(url)
+            .then(response => response.blob())
+            .then((blob) => {
             // Create blob link to download
```
Comment (Jag96) [inline src/libs/fileDownload/index.js:15]: Indent this as well

**Review by Jag96 — CHANGES_REQUESTED** [review]
One thing I noticed on web (chrome)d, if I click the download icon and then click the spinner a few times while the file is downloading, after the first download is done the browser will attempt to download the file additional times based on how many times I clicked the spinner. Can we add a check in `processDownload` that doesn't call `fileDownload` if we are already downloading?

File: src/libs/fileDownload/index.js (hunk 2 above)
Comment (mananjadhav) [reply src/libs/fileDownload/index.js:15]: Add an indent in the second `then` block? I tried adding an indent here and `eslint` automatically removes the indent.

**Comment by mananjadhav** [comment]
@Jag96 Fixed the multiple download issue.  PR updated

**Review by Jag96 — APPROVED** [review]
LGTM

**Pull request merged**